framework/py/flwr/server/superlink/fleet/grpc_rere/node_auth_server_interceptor_test.py

```diff
@@ -293,48 +319,48 @@ def _test_get_fab(self, metadata: list[Any]) -> Any:
         )
         return self._get_fab.with_call(request=req, metadata=metadata)
 
-    def _create_node_in_linkstate(self) -> int:
-        pk_bytes = public_key_to_bytes(self.node_pk)
-        return self.state.create_node(
+    def _create_node_in_linkstate(self, activate: bool = True) -> int:
+        pk_bytes = self.node_pk_bytes
+        node_id = self.state.create_node(
             NOOP_FLWR_AID, public_key=pk_bytes, heartbeat_interval=30
         )
-
-    @parameterized.expand(
-        [
-            (_test_create_node,),
-            (_test_delete_node,),
-            (_test_pull_messages,),
-            (_test_push_messages,),
-            (_test_pull_object,),
-            (_test_push_object,),
-            (_test_get_run,),
-            (_test_send_node_heartbeat,),
-            (_test_get_fab,),
-        ]
-    )  # type: ignore
+        if activate:
+            self.state.activate_node(node_id, 30)
+        return node_id
+
+    rpcs = [
+        (_test_register_node,),
+        (_test_activate_node,),
+        (_test_deactivate_node,),
+        (_test_unregister_node,),
+        (_test_pull_messages,),
+        (_test_push_messages,),
+        (_test_pull_object,),
+        (_test_push_object,),
+        (_test_get_run,),
+        (_test_send_node_heartbeat,),
+        (_test_get_fab,),
+    ]
+
+    @parameterized.expand(rpcs)  # type: ignore
     def test_successful_rpc_with_metadata(
         self, rpc: Callable[[Any, list[Any]], Any]
     ) -> None:
         """Test server interceptor for RPC."""
+        # Skip registration and unregistration tests when node auth is enabled
+        if self.enable_node_auth and rpc.__name__ in [
+            self._test_register_node.__name__,
+            self._test_unregister_node.__name__,
+        ]:
```

Comment:
Should we have a test that checks for a `context.abort` in the scenario when these RPCs are called and node auth is enabled?